packages/nc-gui/components/smartsheet/Grid.vue

```diff
@@ -125,10 +128,65 @@ openNewRecordFormHook?.on(async () => {
 })
 
 const selectCell = (row: number, col: number) => {
+  clearRangeRows()
   selected.row = row
   selected.col = col
 }
 
+const selectBlock = (row: number, col: number) => {
+  if (selected.col === null || selected.row === null) {
+    if(isSelectedBlock){
+      selectedRows.endcol = col;
+      selectedRows.endrow = row;
+    }
+  }else if(selected.col !== col || selected.row !== row){
+    if(isSelectedBlock){
+      selected.col = null;
+      selected.row = null;
+      selectedRows.endcol = col;
+      selectedRows.endrow = row;
+    }
+  }
+}
+
+const selectedRange = (row: number, col: number)=>{
+  if(selectedRows.startrow !== null && selectedRows.startcol !== null && selectedRows.endrow !== null && selectedRows.endcol !== null){
+    rangerows.minrow=selectedRows.startrow > selectedRows.endrow ? selectedRows.endrow : selectedRows.startrow,
+    rangerows.maxrow=selectedRows.startrow < selectedRows.endrow ? selectedRows.endrow : selectedRows.startrow,
+    rangerows.mincol=selectedRows.startcol > selectedRows.endcol ? selectedRows.endcol : selectedRows.startcol,
+    rangerows.maxcol=selectedRows.startcol < selectedRows.endcol ? selectedRows.endcol : selectedRows.startcol
+    return (col>=rangerows.mincol && col<=rangerows.maxcol) && (row>=rangerows.minrow && row<=rangerows.maxrow);
+  }else{
+    return false
+  }
+}
+
+const startSelectRange = (event: MouseEvent, row: number, col: number)=>{
+  if(editEnabled && (selected.col !== col || selected.row !== row)){
+    event.preventDefault();
+  }else if(!editEnabled){
+    event.preventDefault();
+  }
+  selectedRows.startcol = null;
+  selectedRows.endcol = null;
+  selectedRows.startrow = null;
+  selectedRows.endrow = null;
+  selectedRows.startcol = col;
+  selectedRows.startrow = row;
+  isSelectedBlock = true
+}
+
+const clearRangeRows = ()=>{
+  rangerows.mincol = null
+  rangerows.maxcol = null
+  rangerows.minrow = null
+  rangerows.maxrow = null
+  selectedRows.startrow = null
+  selectedRows.startcol = null 
+  selectedRows.endrow = null
+  selectedRows.endcol = null
+}
```

Comment:
- Can this whole block not be simplified?
- If not, add comments so others, especially other contributors, somehow can understand what you meant to do here. Otherwise nobody except yourself can debug this if there are issues
- Possibly outsource this to somewhere else because the Grid.vue file is getting bloated
- Spacing
- No semi colon on EOL